src/sysinfo.cc

```diff
@@ -478,25 +478,27 @@ int GetNumCPUs() {
 #ifdef BENCHMARK_HAS_SYSCTL
   int num_cpu = -1;
   if (GetSysctl("hw.ncpu", &num_cpu)) return num_cpu;
-  fprintf(stderr, "Err: %s\n", strerror(errno));
-  std::exit(EXIT_FAILURE);
+  PrintErrorAndDie("Err: ", strerror(errno));
```

Comment:
This becoming too messy :)
Please split this PR into two:
1. The actual changes to support the architecture
2. All of the PrintErrorAndDie stuff

Reply:
Sorry, a little confused, isn't that what you asked for above?

Reply:
I guess i thought it would look nicer?
Concretely, i'm getting stuck on how asymmetrical this all is,
sometimes there is a check for ncpu=0, sometimes there is not.

My next comment would be to extract the `if ncpu<1 abort` logic
into a wrapper function.

I just don't think it makes sense to do all that cleanup in this PR.

Reply:
OK, will dropping the last commit be sufficient for this PR then?

Reply:
"sysinfo: support parsing /proc/cpuinfo on Alpha" also has some `PrintErrorAndDie` stuff.

Reply:
IMO that is immediately and directly relevant to this PR.  I spent a lot of time investigating and trying to understand why the tests were failing, looking for issues with the cycle timer, when it just turned out that failing to parse `/proc/cpuinfo` silently returns garbage data that results in all sorts of cryptic errors down the line later.  I believe handling this in a sensible way is super important in order to make porting to this and other platforms a reasonable process.

Reply:
Hey, i'm only talking about how to get this diff to the state where i'd be happy to merge it,
and splitting it into two separate chunks will do that.
We spent more time arguing about that than it would have taken to get both of those PR's merged.